processor/intervalprocessor/processor.go

```diff
@@ -102,8 +104,34 @@ func (p *Processor) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) erro
 		rm.ScopeMetrics().RemoveIf(func(sm pmetric.ScopeMetrics) bool {
 			sm.Metrics().RemoveIf(func(m pmetric.Metric) bool {
 				switch m.Type() {
-				case pmetric.MetricTypeGauge, pmetric.MetricTypeSummary:
-					return false
+				case pmetric.MetricTypeSummary:
+					mClone, metricID := p.getOrCloneMetric(rm, sm, m)
+
+					// The ideal scenario is that we would re-use `aggregateDataPoints()` here, but we can't
+					// because the SummaryDataPoint does not fulfill the DataPoint interface.
```

Comment:
it in fact does! all the required methods are there, but the interface is overly restrictive because it has a concrete type list.
it appears to serve no real purpose, because aggregateDataPoints can operate on any type carrying those methods.

I'd suggest removing the type list entirely, making the interface purely behavioral:

```diff
--- a/processor/intervalprocessor/internal/metrics/metrics.go
+++ b/processor/intervalprocessor/internal/metrics/metrics.go
 type DataPoint[Self any] interface {
-       pmetric.NumberDataPoint | pmetric.HistogramDataPoint | pmetric.ExponentialHistogramDataPoint

        Timestamp() pcommon.Timestamp
        Attributes() pcommon.Map
        CopyTo(dest Self)
}
```

Reply:
Yea, I was being overly restrictive. I'm ok with removing the hard type requirements

Reply:
done :)